templates/dashboard/subscription.php

```diff
@@ -7,69 +7,241 @@
-                <form action="" method="post" style="text-align: center;">
-                    <?php wp_nonce_field( 'wpuf-sub-cancel' ); ?>
-                    <input type="hidden" name="gateway" value="<?php echo esc_attr( $payment_gateway ); ?>">
-                    <input type="submit" name="wpuf_cancel_subscription" class="btn btn-sm btn-danger" value="<?php esc_html_e( 'Cancel', 'wp-user-frontend' ); ?>">
-                </form>
-                <?php
+            if ( $user_sub['recurring'] === 'yes' ) {
+                $subscription_data = get_subscription_data( $user_sub );
+                display_subscription_details( $subscription_data );
             }
         }
         ?>
     </div>
 </div>
+
+<?php
+    /**
+     * Get subscription related data
+     *
+     * @param array $user_sub User subscription data
+     * @return array Subscription data
+     */
+function get_subscription_data( $user_sub ) {
+	global $wpdb;
+
+	$user_id = get_current_user_id();
+	$pack_id = $user_sub['pack_id'];
+	$subscription = wpuf()->subscription->get_subscription( $pack_id );
+	// Get payment gateway
+	$payment_gateway = $wpdb->get_var(
+        $wpdb->prepare(
+            "SELECT payment_type 
+            FROM {$wpdb->prefix}wpuf_transaction 
+            WHERE user_id = %d 
+            AND status = 'completed' 
+            ORDER BY created DESC",
+            $user_id
+        )
+    );
+	$payment_gateway = strtolower( $payment_gateway );
+
+	// Get last payment date
+	$last_payment_date = $wpdb->get_var(
+        $wpdb->prepare(
+            "SELECT created 
+            FROM {$wpdb->prefix}wpuf_transaction 
+            WHERE user_id = %d 
+            AND status = 'completed' 
+            ORDER BY created DESC 
+            LIMIT 1",
+            $user_id
+        )
+    );
+
+	// Get billing cycle details
+	$cycle_number = intval( $subscription->meta_value['billing_cycle_number'] );
+	$cycle_period = $subscription->meta_value['cycle_period'];
+
+	// Get trial details
+	$trial_status = $subscription->meta_value['trial_status'];
+	$trial_duration = $subscription->meta_value['trial_duration'];
+	$trial_duration_type = $subscription->meta_value['trial_duration_type'];
+
+	return [
+		'payment_gateway' => $payment_gateway,
+		'last_payment_date' => $last_payment_date,
+		'cycle_number' => $cycle_number,
+		'cycle_period' => $cycle_period,
+		'trial_status' => $trial_status,
+		'trial_duration' => $trial_duration,
+		'trial_duration_type' => $trial_duration_type,
+	];
+}
+
+    /**
+     * Display subscription details
+     *
+     * @param array $subscription_data Subscription data
+     * @return void
+     */
+function display_subscription_details( $subscription_data ) {
+	$trial_html = get_trial_expiration_html( $subscription_data );
+	$billing_html = get_next_billing_html( $subscription_data );
+	?>
+        <br>
+	<?php if ( $subscription_data['trial_status'] == 'on' ) : ?>
```

Comment:
use [Yoda Conditions](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#yoda-conditions) and strict comparisons